### ISSUE TITLE
Pylint is not using a src/ folder

### ISSUE THREAD

**Issue by CarliJoy** [pylint-dev/pylint#5644]
### Bug description

Pylint does not have a src folder

Led me into problems developing and test at least from time to time.

Here an exmplaination stolen why this a good idea:

## Why does PyScaffold ≥ 3 have a ``src`` folder which holds the actual Python package?
   This avoids quite many problems compared to the case when the actual Python package resides in the same folder as
   your configuration and test files.
   A nice [blog post by Ionel](https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure) gives a thorough explanation why this is so. In a nutshell, the most severe
   problem comes from the fact that Python imports a package by first looking at the current working directory and then
   into the ``PYTHONPATH`` environment variable. If your current working directory is the root of your project directory
   you are thus not testing the installation of your package but the local package directly. Eventually, this always
   leads to huge confusion (*"But the unit tests ran perfectly on my machine!"*).

   Moreover, having a dedicated ``src`` directory to store the package files, makes it easy to comply with recent standards
   in the Python community (for example [`PEP 420`](https://www.python.org/dev/peps/pep-0420/)).

   Please notice that PyScaffold assumes all the files inside ``src`` are meant to be part of the package.

### Command used

```shell
cd <other pylint dev folder> && pytest
```

### Pylint output

```shell
some strange stuff, as pytests test the pylint in the other folder while I have also pylint installed as -e from the other -venv
```

### Expected behavior

no problem because src folder prevents them

**Comment by DanielNoord** [comment]
I'm not really a big fan of moving everything around as it will screw up `git blame`. As far as I know we haven't been having any issues with our project structure recently so without any immediate benefit I only see potential downsides here.

-0.5 for me.

**Comment by jacobtylerwalls** [comment]
Same. Major projects like `django` do not use a `src` folder. Thanks for the suggestion, it was worth having a look 👍🏻

**Comment by CarliJoy** [comment]
> I'm not really a big fan of moving everything around as it will screw up `git blame`. As far as I know we haven't been having any issues with our project structure recently so without any immediate benefit I only see potential downsides here.
> 
> -0.5 for me.

FYI @DanielNoord the `git blame` issue can be easily solved with https://akrabat.com/ignoring-revisions-with-git-blame/ 

Also git improved quite a bit with the latest version, so renames are detected much better.

Just to document this.

**Comment by jacobtylerwalls** [comment]
After #8902 and the number of hours sunk into investigating subtle issues, I'm now a fan of doing this. We should at least have a discussion about it.

**Comment by DanielNoord** [comment]
I'm fine with it if we double check that `git blame` still works. If that's the case, let's do it!

**Comment by Pierre-Sassoulas** [comment]
I'm not worried about git blame being broken, the imports should not change, so even small files with import only should be correctly detected as moved not deleted. I'm not using the SRC folder structure in any of my project but I'll get used to it :D

**Comment by DanielNoord** [comment]
I'm also a fan of doing this after having everything come crashing down at my day job because of issues related to this.

Let's do this for the next major release. @CarliJoy Are you still interested in pursuing this and potentially creating a PR? All maintainers seem aligned.

**Comment by CarliJoy** [comment]
@DanielNoord I am sorry I fell victim to `ruff`. So as I am not using pylint anymore. Therefore I am rather investing my time in other issues i.e. into the python typing ecosystem.

But I am happy to help if there are any troubles arising from the migration.

**Comment by DanielNoord** [comment]
That's understandable. I don't think I have time to pick this up myself, but I'd like to thank you for making the original issue :)
